app/models/user.rb

```diff
@@ -270,6 +280,47 @@ def phone_with_extension
     "#{phone}, Ext. #{phone_extension}"
   end
 
+  def assigned_to_lettings_logs
+    lettings_logs.where(assigned_to: self)
+  end
+
+  def assigned_to_sales_logs
+    sales_logs.where(assigned_to: self)
+  end
+
+  def reassign_logs_and_update_organisation(new_organisation, log_reassignment)
+    return unless new_organisation
+
+    ActiveRecord::Base.transaction do
+      lettings_logs_to_reassign = assigned_to_lettings_logs
+      sales_logs_to_reassign = assigned_to_sales_logs
+      current_organisation = organisation
+
+      logs_count = lettings_logs_to_reassign.count + sales_logs_to_reassign.count
+      return if logs_count.positive? && (log_reassignment.blank? || !LOG_REASSIGNMENT.key?(log_reassignment.to_sym))
+
+      update!(organisation: new_organisation)
+
+      case log_reassignment
+      when "reassign_all"
+        reassign_all_orgs(new_organisation, lettings_logs_to_reassign, sales_logs_to_reassign)
+      when "reassign_stock_owner"
+        reassign_stock_owners(new_organisation, lettings_logs_to_reassign, sales_logs_to_reassign)
+      when "reassign_managing_agent"
+        reassign_managing_agents(new_organisation, lettings_logs_to_reassign, sales_logs_to_reassign)
+      when "unassign"
+        unassign_organisations(lettings_logs_to_reassign, sales_logs_to_reassign, current_organisation)
+      end
+
+      send_organisation_change_email(current_organisation, new_organisation, log_reassignment, logs_count)
+    rescue ActiveRecord::RecordInvalid => e
```

Comment:
Are there any other plausible exceptions that we might need to handle?

Reply:
I changed it to StandardError, I think it should be most errors (?) would that work?